seungki/app.js

```diff
@@ -0,0 +1,201 @@
+app.use(cors());
+app.use(morgan('dev'));
+
+const mysqlDatabase = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPEORM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE,
+});
+
+mysqlDatabase
+    .initialize()
+    .then(() => {
+        console.log('Data Source has been inilialized!💡');
+    })
+    .catch((err) => {
+        console.error('Error during Data Source initialization', err);
+        mysqlDatabase.destroy();
+    });
+
+// Health Check
+app.get('/ping', (req, res) => {
+    res.status(200).json({ message: 'pongsssss' });
+});
+
+// 유저 회원가입
+app.post('/email_signup', async (req, res) => {
+    const { name, email, profileImage, password } = req.body;
+
+    await mysqlDatabase.query(
+        `INSERT INTO users(
+      name,
+      email,
+      profile_image,
+      password
+    ) VALUES (?, ?, ?, ?);
+    `,
+        [name, email, profileImage, password]
+    );
+    res.status(201).json({ message: 'userCreated' });
+});
+
+// 게시물 등록하기
+app.post('/post_created', async (req, res) => {
+    const { title, content, postImage, userId } = req.body;
+
+    await mysqlDatabase.query(
+        `INSERT INTO posts(
+      title,
+      content,
+      post_image,
+      user_id
+    ) VALUES (?, ?, ?, ?);
+    `,
+        [title, content, postImage, userId]
+    );
+    res.status(201).json({ message: 'postCreated' });
+});
+
+// 전체 게시글 조회하기
+app.get('/posts', async (req, res) => {
+    await mysqlDatabase.query(
+        `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image AS postingImageUrl,
+      p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p ON u.id = p.user_id`,
+        (err, rows) => {
+            res.status(200).json({ data: rows });
+        }
+    );
+});
+
+// 유저의 게시물 조회하기
+app.get('/user_post/userId/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    const [userPostList] = await mysqlDatabase.query(
+        `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      JSON_ARRAYAGG(
+        JSON_OBJECT(
+          "postingId", p.id,
+          "postingImageUrl", p.post_image,
+          "postingContent", p.content
+        )
+      ) AS postings
+    FROM users u
+    INNER JOIN posts p ON u.id = p.user_id
+    WHERE u.id = ?
+    GROUP BY u.id;
+    `,
+        [userId]
+    );
+    res.status(200).json({ data: userPostList });
+});
+
+// 게시글 수정하기
+app.patch('/post_modify/postId/:postId', async (req, res) => {
+    const { title, content, postImage } = req.body;
+    const { postId } = req.params;
+
+    await mysqlDatabase.query(
+        `UPDATE posts
+      SET title = ?,
+          content = ?,
+          post_image = ?
+      WHERE id = ?
+      `,
+        [title, content, postImage, postId]
+    );
+
```

Comment:
- SQL 문법도 가독성 좋게 만들어주는 것이 중요합니다. 회사마다 컨벤션이 다르겠지만, 위코드에서는 아래와 같은 SQL 문법의 컨벤션을 정하고 있습니다.
- 지금은 위코드에서 사용하고 있는 컨벤션을 지켜서 문법을 작성해주세요. 하지만 이는 회사마다 다르기 때문에, 정답이 아님을 반드시 인지해주세요.
- SQL 포맷팅은 다음 처럼 통일하면 좋을거 같습니다.
```suggestion
// Example
await database.query(`
  UPDATE posts
  SET title = ?,
    description = ?
  WHERE id = ?`,
  [title, description, postId]
);
```